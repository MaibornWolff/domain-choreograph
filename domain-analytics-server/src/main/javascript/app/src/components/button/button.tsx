import * as React from 'react';
import { styled } from '~utils/styled';

export interface ButtonProps {
  lightColor?: boolean;
}

export const Button = styled.button<ButtonProps>`
  border: none;
  padding: .5rem;
  background: none;
  font-size: 1rem;
  cursor: pointer;
  border-radius: ${({theme}) => theme.dimensions.borderRadius.normal};
  color: ${({ lightColor, theme }) =>
  lightColor
    ? theme.colors.text.light.primary
    : theme.colors.text.dark.primary
  };

  &:hover, &:focus {
    outline: none;
    background: rgba(0, 0, 0, .15);
  }
`;
