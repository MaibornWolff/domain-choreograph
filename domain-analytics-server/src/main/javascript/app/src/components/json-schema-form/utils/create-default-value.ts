import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';

const defaultValues: Record<Exclude<JSONSchema4TypeName, 'object'>, any> = {
  string: undefined,
  number: undefined,
  integer: undefined,
  boolean: undefined,
  array: [],
  'null': null,
  any: ''
};

export function createDefaultValue(schema: JSONSchema4): any {
  const type = schema.type || 'any';
  if (type instanceof Array) {
    throw new Error('Array types are not supported');
  } else if (type === 'object') {
    return createDefaultValueForObject(schema);
  } else {
    return defaultValues[type];
  }
}

function createDefaultValueForObject(schema: JSONSchema4): Promise<object> {
  if (schema.type !== 'object') {
    throw new Error('Expect schema to be an object');
  }
  let defaultValue: any = {};
  if (schema.properties) {
    Object.entries(schema.properties).forEach(([key, property]) => {
      defaultValue[key] = createDefaultValue(property)
    });
  }
  return defaultValue;
}

