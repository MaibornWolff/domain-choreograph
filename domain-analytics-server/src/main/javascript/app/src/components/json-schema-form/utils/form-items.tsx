import { styled } from '~utils/styled';

export const FormItems = styled('div')`
    display: flex;
    flex-direction: column;

    & > * {
      margin-bottom: .5rem;
    }
`;
