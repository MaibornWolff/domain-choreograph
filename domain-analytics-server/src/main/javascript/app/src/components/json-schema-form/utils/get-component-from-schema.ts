import { FieldComponent } from 'clean-forms';
import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
import { DatePickerMsField } from '~components/json-schema-form/form-fields/date-picker-ms-field';
import { NumberTextField, TextField } from '~components/json-schema-form/form-fields/text-field';
import { JsonSchemaFormFieldProps } from '~components/json-schema-form/models/json-schema-form-field-props';

export interface JsonSchemaComponentMap {
  matcher?: { type: JSONSchema4TypeName, format?: string };
  field: FieldComponent<any, JsonSchemaFormFieldProps>;
}

const defaultMapping: JsonSchemaComponentMap[] = [
  { matcher: { type: 'integer', format: 'utc-millisec' }, field: DatePickerMsField },
  { matcher: { type: 'integer' }, field: NumberTextField },
  { matcher: { type: 'number' }, field: NumberTextField },
  { field: TextField }
];

export function getComponentFromSchema(
  schema: JSONSchema4,
  mapping: JsonSchemaComponentMap[] = defaultMapping
): FieldComponent<any, JsonSchemaFormFieldProps> {
  for (const map of mapping) {
    if (map.matcher == null) {
      return map.field;
    }
    const typeIsMatching = map.matcher.type === schema.type;
    const formatIsMatching = map.matcher.format == null || map.matcher.format === schema.format;
    if (typeIsMatching && formatIsMatching) {
      return map.field;
    }
  }
  throw new Error('Couldn\'t match the schema');
}
