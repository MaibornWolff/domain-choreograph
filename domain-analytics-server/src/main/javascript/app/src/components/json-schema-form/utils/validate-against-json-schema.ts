import { ValidationFunction } from 'clean-forms';
import { JSONSchema4 } from 'json-schema';
import { Validator } from 'jsonschema';


export function validateAgainstJsonSchema(schema: JSONSchema4): ValidationFunction {
  const validator = new Validator();
  return ({ value }) => {
    const result = validator.validate(value, schema as any);
    const errors = result.errors.map(error => {
      let message = error.message;
      let property = error.property
        .replace(/^instance\.?/, '');
      let schema = error.schema as JSONSchema4;
      if (error.name === 'required') {
        message = `is required`;
        schema = schema.properties![error.argument];
        property = joinProperties(property, error.argument);
      }

      const title = schema.title || '';
      message = `${title} ${message}`;

      return [ property, message ] as [string, string];
    });
    return errors;
  }
}

function joinProperties(path1: string, path2: string): string {
  if (path1 === '') {
    return path2;
  }
  if (path2 === '') {
    return path1;
  }
  return `${path1}.${path2}`;
}
