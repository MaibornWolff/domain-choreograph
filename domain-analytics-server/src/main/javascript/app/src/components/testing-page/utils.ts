export function escapeJavaClass(javaClass: string) {
  return javaClass.replace(/\./g, '-');
}

export function reverseJavaClassEscape(javaClass: string) {
  return javaClass.replace(/-/g, '.');
}
