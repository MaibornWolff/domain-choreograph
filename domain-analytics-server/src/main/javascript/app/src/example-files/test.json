{
  "nodes": {
    "A": {
      "id": "A",
      "name": "A",
      "scope": "root"
    }
  },
  "dependencies": [],
  "scopes": {
    "root": {
      "id": "root",
      "executionContext": "Application",
      "nodes": [
        "A"
      ]
    }
  },
  "executionContexts": {
    "Application": {
      "id": "Application",
      "scopes": [
        "root"
      ]
    }
  }
}
