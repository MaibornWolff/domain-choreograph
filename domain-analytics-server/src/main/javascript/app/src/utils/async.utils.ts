export function wait(milliseconds: number): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    setTimeout(() => resolve(), milliseconds);
  });
}
