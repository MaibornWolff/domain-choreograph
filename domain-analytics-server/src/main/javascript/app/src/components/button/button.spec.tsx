import * as React from 'react';
import { renderWithProviders } from '~utils/test.utils';
import { Button } from './button';

describe('Button', () => {
  it('should render', () => {
    expect(renderWithProviders(<Button/>)).toMatchSnapshot();
  });

  it('should render with light font', () => {
    expect(renderWithProviders(<Button lightColor={true}/>)).toMatchSnapshot();
  });
});
