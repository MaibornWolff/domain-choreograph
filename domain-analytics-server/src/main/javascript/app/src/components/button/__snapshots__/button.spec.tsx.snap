// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.emotion-0 {
  border: none;
  padding: .5rem;
  background: none;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 3px;
  color: rgba(0,0,0,.87);
}

.emotion-0:hover,
.emotion-0:focus {
  outline: none;
  background: rgba(0,0,0,.15);
}

<button
  class="emotion-0 emotion-1"
/>
`;

exports[`Button should render with light font 1`] = `
.emotion-0 {
  border: none;
  padding: .5rem;
  background: none;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 3px;
  color: rgba(255,255,255,1);
}

.emotion-0:hover,
.emotion-0:focus {
  outline: none;
  background: rgba(0,0,0,.15);
}

<button
  class="emotion-0 emotion-1"
/>
`;
